javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrinter.java

```diff
@@ -21,41 +21,60 @@
 
 package com.github.javaparser.printer;
 
+import java.util.function.Function;
+
 import com.github.javaparser.ast.Node;
 import com.github.javaparser.ast.visitor.VoidVisitor;
+import com.github.javaparser.printer.configuration.ConfigurablePrinter;
+import com.github.javaparser.printer.configuration.PrettyPrinterConfiguration;
 
 /**
  * Pretty printer for AST nodes.
+ * This class is no longer acceptable to use because it is not sufficiently configurable and it is too tied to a specific implementation
+ * <p> Use {@link Printable interface or PrettyPrintable default implementation } instead.
  */
+@Deprecated
```

Comment:
This class was added so that people could easily find out how to do pretty printing. Maybe it is no longer wanted, but it would be wise to keep the name "PrettyPrinter" since that is the first thing users will look for when looking for pretty printing functionality.

Reply:
I agree with you and this class is still usable for compatibility purpose. It's why i propose to deprecate this class for the moment. Perhaps this comment "This class is no longer acceptable to use because..." is misinterpreted or poorly worded. I can update this comment if it's the case.